samcli/lib/deploy/deployer.py

```diff
@@ -314,19 +314,22 @@ def wait_for_changeset(self, changeset_id, stack_name):
             waiter.wait(ChangeSetName=changeset_id, StackName=stack_name, WaiterConfig=waiter_config)
         except botocore.exceptions.WaiterError as ex:
             resp = ex.last_response
-            status = resp["Status"]
-            reason = resp["StatusReason"]
-
-            if (
-                status == "FAILED"
-                and "The submitted information didn't contain changes." in reason
-                or "No updates are to be performed" in reason
-            ):
-                raise deploy_exceptions.ChangeEmptyError(stack_name=stack_name)
-
-            raise ChangeSetError(
-                stack_name=stack_name, msg="ex: {0} Status: {1}. Reason: {2}".format(ex, status, reason)
-            ) from ex
+            if "Status" in resp and "StatusReason" in resp:
+                status = resp["Status"]
+                reason = resp["StatusReason"]
```

Comment:
If `Status` is not in the response, what does that indicate? 

I ask because I typically push to avoid checks like this, especially on dicts. It's a better practice to `resp.get("Status", <sane default>)`, which allows for lest nesting and easier to read/maintain code.

Reply:
If `Status` and `StatusReason` are missing, it means the response is not a valid CloudFormation `DescribeChangeSet` response, and could be something else like Credentials error. 

Some form of branching is still needed to handle the error differently IMO. But let me try to write it in a less nested manner.